test/e2e/azure_logcollector.go

```diff
@@ -50,40 +52,61 @@ func (k AzureLogCollector) CollectMachineLog(ctx context.Context, managementClus
+	cluster, err := util.GetClusterFromMetadata(ctx, managementClusterClient, m.ObjectMeta)
+	if err != nil {
+		return err
+	}
+
+	hostname := getHostname(m, isAzureMachineWindows(am))
+
+	if err := collectLogsFromNode(ctx, managementClusterClient, cluster, hostname, isAzureMachineWindows(am), outputPath); err != nil {
 		errors = append(errors, err)
 	}
 
-	if err := collectBootLog(ctx, am, outputPath); err != nil {
+	if err := collectVMBootLog(ctx, am, outputPath); err != nil {
 		errors = append(errors, err)
 	}
 
 	return kinderrors.NewAggregate(errors)
-
 }
 
-// collectLogsFromNode collects logs from various sources by ssh'ing into the node
-func collectLogsFromNode(ctx context.Context, managementClusterClient client.Client, m *clusterv1.Machine, am *v1alpha4.AzureMachine, outputPath string) error {
-	cluster, err := util.GetClusterFromMetadata(ctx, managementClusterClient, m.ObjectMeta)
+// CollectMachinePoolLog collects logs from a machine pool.
+func (k AzureLogCollector) CollectMachinePoolLog(ctx context.Context, managementClusterClient client.Client, mp *expv1.MachinePool, outputPath string) error {
+	var errors []error
+
+	am, err := getAzureMachinePool(ctx, managementClusterClient, mp)
 	if err != nil {
 		return err
 	}
 
-	Logf("INFO: Collecting logs for machine %s in cluster %s in namespace %s\n", m.GetName(), cluster.Name, cluster.Namespace)
-	isWindows := isNodeWindows(am)
+	cluster, err := util.GetClusterFromMetadata(ctx, managementClusterClient, mp.ObjectMeta)
+	if err != nil {
+		return err
+	}
 
-	controlPlaneEndpoint := cluster.Spec.ControlPlaneEndpoint.Host
-	hostname := m.Spec.InfrastructureRef.Name
-	if isWindows {
-		// Windows host name ends up being different than the infra machine name
-		// due to Windows name limitations in Azure so use ipaddress instead
-		if len(m.Status.Addresses) > 0 {
-			hostname = m.Status.Addresses[0].Address
-		} else {
-			Logf("INFO: Unable to collect logs as node doesn't have addresses")
+	isWindows := isAzureMachinePoolWindows(am)
+
+	for i, instance := range mp.Spec.ProviderIDList {
+		hostname := mp.Status.NodeRefs[i].Name
+
+		if err := collectLogsFromNode(ctx, managementClusterClient, cluster, hostname, isWindows, filepath.Join(outputPath, hostname)); err != nil {
+			errors = append(errors, err)
+		}
+
+		if err := collectVMSSBootLog(ctx, instance, filepath.Join(outputPath, hostname)); err != nil {
+			errors = append(errors, err)
 		}
 	}
 
+	return kinderrors.NewAggregate(errors)
+}
+
+// collectLogsFromNode collects logs from various sources by ssh'ing into the node
+func collectLogsFromNode(ctx context.Context, managementClusterClient client.Client, cluster *clusterv1.Cluster, hostname string, isWindows bool, outputPath string) error {
+	Logf("INFO: Collecting logs for node %s in cluster %s in namespace %s\n", hostname, cluster.Name, cluster.Namespace)
+
+	controlPlaneEndpoint := cluster.Spec.ControlPlaneEndpoint.Host
+
 	port := e2eConfig.GetVariable(VMSSHPort)
```

Comment:
Reading `VMSSHPORT` I thought this was for for Machine sets (VMSS) and was confused for a few since it was refactored it to work for both vmss and machines. For readability it might be a nice change to `sshPort`

Reply:
fixed, removed the var and made it a const